wandb/sdk/wandb_run.py

```diff
@@ -2232,13 +2233,19 @@ def _public_api(self) -> PublicApi:
     # TODO(jhr): annotate this
     def _assert_can_log_artifact(self, artifact) -> None:  # type: ignore
         if not self._settings._offline:
-            public_api = self._public_api()
-            expected_type = public.Artifact.expected_type(
-                public_api.client,
-                artifact.name,
-                public_api.settings["entity"],
-                public_api.settings["project"],
-            )
+            try:
+                public_api = self._public_api()
+                expected_type = public.Artifact.expected_type(
+                    public_api.client,
+                    artifact.name,
+                    public_api.settings["entity"],
+                    public_api.settings["project"],
+                )
+            except requests.exceptions.ConnectionError:
```

Comment:
I think we want `requests.RequestException` to handle 500's or non connection related errors.

Reply:
Updated - thanks!